daffodil-tdml-lib/src/main/scala/org/apache/daffodil/tdml/TDMLRunner.scala

```diff
@@ -2360,6 +2354,45 @@ sealed abstract class DocumentPart(part: Node, parent: Document) {
 
 }
 
+object CanonData {
+  private lazy val doubleForwardPattern = "//.*".r
+  private lazy val openClosePattern = "(?s)/[*].*?[*]/".r
+  private lazy val noWarnCharsSet = "|()[].Xx \n"
+
+  /*
+  * Allow "//" and "/* */" to act as comments.
+  * Any valid XML characters not explicitly allowed are also considered comments and are removed.
+  */
+  def canonicalizeData(validCharactersSet : String, userData : String) : String = {
+    var doWarning: Boolean = false
+    Assert.invariant(!userData.contains('\r'))      // \r should not exist in userData
+
+    //Remove the comments (//) and (/* */)
+    val noCommentsForward = doubleForwardPattern.replaceAllIn(userData,"")
+    val noCommentsBothFormats = openClosePattern.replaceAllIn(noCommentsForward,"")
+
+    //Throw exception if /* or */ still found. This means user input was not formatted correctly.
+    if (noCommentsBothFormats.contains("/*") || noCommentsBothFormats.contains("*/")) {
+      throw TDMLException("Improper formatting of /* */ style comment", None)
+    }
+
+    //Check value of the characters, if invalid character found create log and skip over it
+    val validData = noCommentsBothFormats.filter { ch =>
+      if (validCharactersSet.contains(ch)) true
+      else {
+        if (!noWarnCharsSet.contains(ch)) doWarning = true
+        false
+      }
+    }
+
+    if (doWarning) {
+        Logger.log.warn("Data contains invalid character(s). Consider using a comment (// or /* */).")
+    }
+
+    return validData
```

Comment:
There shouldn't be a `return` keyword here. Just simply `validData` as this line is the scala standard.